crates/runtime/src/dataconnector/s3.rs

```diff
@@ -124,6 +129,52 @@ impl S3 {
 
         Ok(s3_url)
     }
+
+    fn get_file_format_and_extension(&self) -> AnyErrorResult<(Arc<dyn FileFormat>, String)> {
+        let params = &self.params;
+        match params.get("file_format") {
```

Comment:
This code is very nested - is there a way to improve it? I think just creating separate functions for handling the CSV and Parquet match arms would be enough.